graphql/util.go

```diff
@@ -427,46 +427,37 @@ func mapHTTPStatusToGqlError(ctx context.Context, httpStatus int, err error) *gq
 	}
 }
 
-func isTaskBlocked(ctx context.Context, at *restModel.APITask) (*bool, error) {
-	t, err := task.FindOneIdNewOrOld(*at.Id)
-	if err != nil {
-		return nil, ResourceNotFound.Send(ctx, err.Error())
+func canRestartTask(t *task.Task) bool {
+	// Cannot restart execution tasks.
+	isExecTask := t.IsPartOfDisplay()
+	if isExecTask {
+		return false
 	}
-	if t == nil {
-		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("task %s not found", *at.Id))
+	if !utility.StringSliceContains(evergreen.TaskUncompletedStatuses, t.Status) {
+		return true
 	}
-	isBlocked := t.Blocked()
-	return &isBlocked, nil
-}
-
-func isExecutionTask(ctx context.Context, at *restModel.APITask) (*bool, error) {
-	i, err := at.ToService()
-	if err != nil {
-		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error while converting task %s to service", *at.Id))
+	if t.Aborted {
```

Comment:
I would suggest moving this case to the bottom and just returning !t.Aborted (as opposed to return false) to free up an if statement